src/Account/Session/ShopSession.php

```diff
@@ -78,6 +85,11 @@ public function getOrRefreshToken($forceRefresh = false)
     public function refreshToken($refreshToken = null)
     {
         try {
+            if (!$this->oauth2ClientProvider->getOauth2Client()->exists()) {
+                $this->linkShop->delete();
+                $this->configurationRepository->clearLinkContext();
```

Comment:
maybe we should trigger an UnlinkShopCommand here which already do the local cleanup the right way.

Reply:
Yes actually we though about it during the pair coding but you said there wasn't a clean of everything like in the description of the issue i think :thinking:

Reply:
after looking further that's not the case, the command cleanup link context triggering a dedicated hook

Reply:
Okay, i should so change all the code written and the test so?

Reply:
yes, something like that should do the job :
`$this->commandBus->handle(new UnlinkShopCommand($shop->id));`